broker/broker.go

```diff
@@ -212,15 +220,19 @@ func clientOffers(ctx *BrokerContext, w http.ResponseWriter, r *http.Request) {
 		return
 	}
 	// Immediately fail if there are no snowflakes available.
-	if ctx.snowflakes.Len() <= 0 {
+	ctx.snowflakeLock.Lock()
+	numSnowflakes := ctx.snowflakes.Len()
+	ctx.snowflakeLock.Unlock()
+	if numSnowflakes <= 0 {
 		ctx.metrics.clientDeniedCount++
```

Comment:
Should this increment also be protected by a lock? Or is it impossible for two goroutines to execute it simultaneously?

Reply:
Thanks for pointing this out. There is a chance for a race here if the counts are updated at the same time that the metrics are being logged (this happens once every 24 hours).

I added a different lock to the metrics struct to handle this in a new commit. How does it look?